web/utils.py

```
@@ -414,6 +416,288 @@ def create_leaderboard_context(
+
+
+def get_stripe_customer(user: "User") -> Optional["stripe.Customer"]:
+    """
+    Retrieve or create a Stripe customer for the given user.
+    """
+    setup_stripe()
+    try:
+        if not user.membership.stripe_customer_id:
+            # Create a new customer
+            customer = stripe.Customer.create(
+                email=user.email, name=f"{user.first_name} {user.last_name}".strip(), metadata={"user_id": user.id}
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)